Makefile

```diff
@@ -1,123 +1,104 @@
-# Metadata about this makefile and position
-MKFILE_PATH := $(lastword $(MAKEFILE_LIST))
-CURRENT_DIR := $(patsubst %/,%,$(dir $(realpath $(MKFILE_PATH))))
+SHELL := /bin/bash -o pipefail
 
-# Ensure GOPATH
-GOPATH ?= $(HOME)/go
+# List of tests to run
+FILES ?= ./...
 
 # List all our actual files, excluding vendor
-GOFILES ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOPKGS ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOFILES ?= $(shell find . -name '*.go' | grep -v /vendor/)
 
 # Tags specific for building
 GOTAGS ?=
 
 # Number of procs to use
 GOMAXPROCS ?= 4
 
-PROJECT := $(CURRENT_DIR:$(GOPATH)/src/%=%)
-OWNER := $(notdir $(patsubst %/,%,$(dir $(PROJECT))))
-NAME := $(notdir $(PROJECT))
-EXTERNAL_TOOLS = \
-	github.com/golang/dep/cmd/dep
-
-# Current system information
-GOOS ?= $(shell go env GOOS)
-GOARCH ?= $(shell go env GOARCH)
-GOCACHE ?= $(shell go env GOCACHE)
-
-# List of tests to run
-FILES ?= ./...
+NAME := $(notdir $(shell pwd))
 
 # Test Service ID
 FASTLY_TEST_SERVICE_ID ?=
 
-# bootstrap installs the necessary go tools for development or build.
-bootstrap:
-	@echo "==> Bootstrapping ${PROJECT}"
-	@for t in ${EXTERNAL_TOOLS}; do \
-		echo "--> Installing $$t" ; \
-		go get -u "$$t"; \
-	done
-.PHONY: bootstrap
-
-clean:
-	@echo "==> Cleaning ${NAME}"
-	@rm -rf pkg
-.PHONY: clean
-
-# build builds the binary into pkg/
-build:
-	@echo "==> Building ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go build -a -o "pkg/${GOOS}_${GOARCH}/${NAME}" ${GOFILES}
-.PHONY: build
-
-# deps updates all dependencies for this project.
-deps:
-	@echo "==> Updating deps for ${PROJECT}"
-	@dep ensure -update
-	@dep prune
-.PHONY: deps
-
-# dev builds and installs the
-dev:
-	@echo "==> Installing ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go install ${GOFILES}
-.PHONY: dev
-
-# linux builds the linux binary
-linux:
-	@env \
-		GOOS="linux" \
-		GOARCH="amd64" \
-		$(MAKE) -f "${MKFILE_PATH}" build
-.PHONY: linux
+all: mod-download dev-dependencies tidy fmt fiximports test vet staticcheck ## Runs all of the required cleaning and verification targets.
+.PHONY: all
+
+tidy: ## Cleans the Go module.
+	@echo "==> Tidying module"
+	@go mod tidy
+.PHONY: tidy
 
-# test runs the test suite.
-test:
+mod-download: ## Downloads the Go module.
+	@echo "==> Downloading Go module"
+	@go mod download
+.PHONY: mod-download
+
+dev-dependencies: ## Downloads the necessesary dev dependencies.
+	@echo "==> Downloading development dependencies"
+	@go install honnef.co/go/tools/cmd/staticcheck
+	@go install golang.org/x/tools/cmd/goimports
+.PHONY: dev-dependencies
```

Comment:
Doesn't this dirty the `go.mod` `go.sum` manifests?

Reply:
Yes, it does add to the Module manifest files. But I wouldn't consider this "dirtying" them. So the goal of adding them to the module is to ensure that developers are using the same version of the dependency.

For example, you wouldn't want someone running a super old version of staticcheck that doesn't yet include new, valuable, checks.

Reply:
Someone shared https://github.com/bwplotka/bingo internally and I did look into it and it does solve shortcomings of the current advice of the community (i.e., `package tools` approach), namely:

1. Avoids dependency and tools defining different constraints on transitive dependencies.
2. Tools aren't included as part of the `go-fastly` module i.e., what package consumers are interested in.

I don't have strong opinions, but in my opinion, it feels like a hacky solution built on top of an already hacky solution. Unless we are concerned with package consumers receiving tool dependencies and until we face point \#1, described above, I don't think the added complexity of another tool and managing sub-modules is worth it.